Twilio-org/phonebank
# Add Join Button to Volunteer Page and Refactor User Actions & Action Tests

Description:
- Create addCampaignToUser action
- Hook up Join Button on 'volunteers/campaigns' page with addCampaignToUser action
  - Update list_view "setCurrentItem" prop to "componentClickHandler" to be more explicit and put placeholder functions for list views that don't handle clicks yet.
- Refactor account_info, edit_account, and registration actions files into one users.jsx action file
- Refactor account_info, edit_account, and registration action tests into one users.test.js file
- Rename account_info reducer to users.jsx
- Update store and other files to import the new users file path

Notes:
- Will probably want addCampaignToUser to dispatch an action in order to update the state with the newly joined campaign. Then, we can use the updated state to reformat the Join button. I believe this action is a part of another open PR.

Test:
- run `npm run test:client users.test.js`
- run `npm run db:seed` file, register a new user, click "Join" button next to "Campaign 1 Title"
- run `npm run db` and `select * FROM campaigns_users WHERE user_id=<user_id to check>;`
   - make sure the campaign_user association exists

Jira Ticket 87
